benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -302,18 +326,72 @@ void GraphicsBenchmarkApp::SetupFullscreenQuadsResources()
+        layoutCreateInfo.bindings.push_back(grfx::DescriptorBinding(QUADS_SAMPLED_IMAGE_REGISTER, grfx::DESCRIPTOR_TYPE_SAMPLED_IMAGE));
         PPX_CHECKED_CALL(GetDevice()->CreateDescriptorSetLayout(&layoutCreateInfo, &mFullscreenQuads.descriptorSetLayout));
     }
 
+    // Allocate descriptor sets
+    uint32_t n = GetNumFramesInFlight();
+    for (size_t i = 0; i < n; i++) {
+        grfx::DescriptorSetPtr pDescriptorSet;
+        PPX_CHECKED_CALL(GetDevice()->AllocateDescriptorSet(mDescriptorPool, mFullscreenQuads.descriptorSetLayout, &pDescriptorSet));
+        mFullscreenQuads.descriptorSets.push_back(pDescriptorSet);
+    }
+
+    UpdateFullscreenQuadsDescriptors();
+
     SetupShader("Benchmark_VsSimpleQuads.vs", &mVSQuads);
     SetupShader("Benchmark_RandomNoise.ps", &mQuadsPs[0]);
     SetupShader("Benchmark_SolidColor.ps", &mQuadsPs[1]);
     SetupShader("Benchmark_Texture.ps", &mQuadsPs[2]);
 }
 
-void GraphicsBenchmarkApp::SetupSkyboxMeshes()
+void GraphicsBenchmarkApp::UpdateSkyBoxDescriptors()
+{
+    uint32_t n = GetNumFramesInFlight();
+    for (size_t i = 0; i < n; i++) {
+        grfx::DescriptorSetPtr pDescriptorSet = mSkyBox.descriptorSets[i];
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateUniformBuffer(SKYBOX_UNIFORM_BUFFER_REGISTER, 0, mSkyBox.uniformBuffer));
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SKYBOX_SAMPLED_IMAGE_REGISTER, 0, mSkyBoxTexture));
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateSampler(SKYBOX_SAMPLER_REGISTER, 0, mLinearSampler));
+    }
+}
+
+void GraphicsBenchmarkApp::UpdateSphereDescriptors()
+{
+    uint32_t n = GetNumFramesInFlight();
+    for (size_t i = 0; i < n; i++) {
+        grfx::DescriptorSetPtr pDescriptorSet = mSphere.descriptorSets[i];
+
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateUniformBuffer(SPHERE_UNIFORM_BUFFER_REGISTER, 0, mSphere.uniformBuffer));
+
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateSampler(SPHERE_ALBEDO_SAMPLER_REGISTER, 0, mLinearSampler));
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateSampler(SPHERE_NORMAL_SAMPLER_REGISTER, 0, mLinearSampler));
+        PPX_CHECKED_CALL(pDescriptorSet->UpdateSampler(SPHERE_METAL_ROUGHNESS_SAMPLER_REGISTER, 0, mLinearSampler));
+
+        if (pAllTexturesTo1x1->GetValue()) {
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_ALBEDO_SAMPLED_IMAGE_REGISTER, 0, mWhitePixelTexture));
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_NORMAL_SAMPLED_IMAGE_REGISTER, 0, mWhitePixelTexture));
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_METAL_ROUGHNESS_SAMPLED_IMAGE_REGISTER, 0, mWhitePixelTexture));
+        }
+        else {
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_ALBEDO_SAMPLED_IMAGE_REGISTER, 0, mAlbedoTexture));
+            PPX_CHECKED_CALL(pDescriptorSet->UpdateSampledImage(SPHERE_NORMAL_SAMPLED_IMAGE_REGISTER, 0, mAlbedoTexture));
```

Comment:
mNormalMapTexture

Reply:
Oops, thanks for catching this. I was trying to figure out why the spheres looked different.